spark/src/main/scala/org/apache/spark/sql/delta/IcebergCompat.scala

```diff
@@ -398,6 +399,29 @@ object CheckTypeInV2AllowList extends IcebergCompatCheck {
   }
 }
 
+object CheckPartitionDataTypeInV2AllowList extends IcebergCompatCheck {
+  private val allowTypes = Set[Class[_]] (
+    ByteType.getClass, ShortType.getClass, IntegerType.getClass, LongType.getClass,
+    FloatType.getClass, DoubleType.getClass, classOf[DecimalType],
+    StringType.getClass, BinaryType.getClass,
+    BooleanType.getClass,
+    TimestampType.getClass, TimestampNTZType.getClass, DateType.getClass
+  )
+  override def apply(context: IcebergCompatContext): Unit = {
+    val partitionSchema = context.newestMetadata.partitionSchema
+    if (partitionSchema.nonEmpty) {
```

Comment:
Actually this check isn't quite right, I'll need to fix this and make sure tests exercise it correctly.